kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -133,15 +133,35 @@ public void flush(Map<TopicPartition, OffsetAndMetadata> offsets) {
     topicPartitionManager.resumeAll();
   }
 
+  private void maybeEnsureExistingTable(TableId table) {
+    BigQuery bigQuery = getBigQuery();
+    if (bigQuery.getTable(table) == null && !config.getBoolean(config.TABLE_CREATE_CONFIG)) {
+      logger.warn("You may want to enable auto table creation by setting {}=true in the properties file",
+              config.TABLE_CREATE_CONFIG);
+      throw new BigQueryConnectException("Table '" + table + "' does not exist");
+    }
+  }
+
   private PartitionedTableId getRecordTable(SinkRecord record) {
-    // Dynamically update topicToBaseTableIds mapping. topicToBaseTableIds was used to be
-    // constructed when connector starts hence new topic configuration needed connector to restart.
-    // Dynamic update shall not require connector restart and shall compute table id in runtime.
-    if (!topicsToBaseTableIds.containsKey(record.topic())) {
-      TopicToTableResolver.updateTopicToTable(config, record.topic(), topicsToBaseTableIds);
+    String tableName;
+    String dataset = config.getString(config.DEFAULT_DATASET_CONFIG);
+    String[] smtReplacement = record.topic().split(":");
+
+    if (smtReplacement.length == 2) {
+      dataset = smtReplacement[0];
+      tableName = smtReplacement[1];
+    } else if (smtReplacement.length == 1) {
+      tableName = smtReplacement[0];
+    } else {
+      throw new ConfigException("Incorrect regex replacement format. " +
+              "SMT replacement should either follow <dataset>:<tableName> format or replace by <tableName> only.");
```

Comment:
This message could be a bit unclear. I think it'd be clearer like
`SMT replacement should either produce the <dataset>:<tableName> format or just the <tablename> format`.